torchvision/models/regnet.py

```diff
@@ -0,0 +1,708 @@
+        activation_layer: Callable[..., nn.Module],
+        group_width: int = 1,
+        bottleneck_multiplier: float = 4.0,
+        se_ratio: Optional[float] = None,
+    ) -> None:
+        super().__init__()
+        self.has_skip = (width_in == width_out) and (stride == 1)
+        self.f = BottleneckTransform(
+            width_in,
+            width_out,
+            stride,
+            norm_layer,
+            activation_layer,
+            group_width,
+            bottleneck_multiplier,
+            se_ratio,
+        )
+
+    def forward(self, x: Tensor) -> Tensor:
+        return x + self.f(x) if self.has_skip else self.f(x)
+
+
+class AnyStage(nn.Sequential):
+    """AnyNet stage (sequence of blocks w/ the same output shape)."""
+
+    def __init__(
+        self,
+        width_in: int,
+        width_out: int,
+        stride: int,
+        depth: int,
+        block_constructor: Callable[..., nn.Module],
+        norm_layer: Callable[..., nn.Module],
+        activation_layer: Callable[..., nn.Module],
+        group_width: int,
+        bottleneck_multiplier: float,
+        se_ratio: Optional[float] = None,
+        stage_index: int = 0,
+    ) -> None:
+        super().__init__()
+
+        for i in range(depth):
+            block = block_constructor(
+                width_in if i == 0 else width_out,
+                width_out,
+                stride if i == 0 else 1,
+                norm_layer,
+                activation_layer,
+                group_width,
+                bottleneck_multiplier,
+                se_ratio,
+            )
+
+            self.add_module(f"block{stage_index}-{i}", block)
+
+
+class RegNetParams:
+    def __init__(
+        self,
+        depth: int,
+        w_0: int,
+        w_a: float,
+        w_m: float,
+        group_width: int,
+        bottleneck_multiplier: float = 1.0,
+        stem_type: Callable[..., nn.Module] = SimpleStemIN,
+        stem_width: int = 32,
+        block_type: Callable[..., nn.Module] = ResBottleneckBlock,
+        activation: Callable[..., nn.Module] = nn.ReLU,
+        use_se: bool = True,
+        se_ratio: float = 0.25,
+        bn_epsilon: float = 1e-05,
+        bn_momentum: float = 0.1,
+        num_classes: int = 1000,
+    ) -> None:
+        if w_a < 0 or w_0 <= 0 or w_m <= 1 or w_0 % 8 != 0:
+            raise ValueError("Invalid RegNet settings")
+        self.depth = depth
+        self.w_0 = w_0
+        self.w_a = w_a
+        self.w_m = w_m
+        self.group_width = group_width
+        self.bottleneck_multiplier = bottleneck_multiplier
+        self.stem_type = stem_type
+        self.block_type = block_type
+        self.activation = activation
+        self.norm_layer = partial(nn.BatchNorm2d, eps=bn_epsilon, momentum=bn_momentum)
+        self.stem_width = stem_width
+        self.use_se = use_se
+        self.se_ratio = se_ratio if use_se else None
+        self.num_classes = num_classes
+
+    def get_expanded_params(self):
```

Comment:
Reason to keep this API in `RegNetParams`:
Here we calculate the width and depth for each block in `get_expanded_params`, but In model scaling use case, we want to pass in pre-computed blocks widths, depths as Lists. See [AnyNet in Classy](https://github.com/facebookresearch/ClassyVision/blob/main/classy_vision/models/anynet.py#L367-L371). I'm thinking for model scaling, user could inherit `RegNetParams` and override the `get_expanded_params` method to directly return the block settings.
cc @mannatsingh

Reply:
I think something like this is quite important! But also, if we want to support it, we should do it fully instead of allowing users to override it with their code - this can get messy.

Reply:
@mannatsingh alternatively, we could change the constructor to take list of depths, widths, group_widths etc. and have a class method that take `w_0`, `w_a`, `w_m` to compute the lists.

Reply:
I would probably make this private:
```suggestion
    def _get_expanded_params(self):
```